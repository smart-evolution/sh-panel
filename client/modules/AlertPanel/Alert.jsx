import React from 'react';
import PropTypes from 'prop-types';

class Alert extends React.Component {
  constructor() {
    super();

    this.state = {
      show: false,
      hide: false,
    };
  }

  componentDidMount() {
    this.showTimeout = setTimeout(() => {
      this.setState({ show: true });
    }, 2000);

    this.hideTimeout = setTimeout(() => {
      this.setState({ hide: true });
    }, 4000);
  }

  componentWillUnmount() {
    if (this.showTimeout) {
      clearTimeout(this.showTimeout);
    }
    if (this.hideTimeout) {
      clearTimeout(this.hideTimeout);
    }
  }

  render() {
    const { type, children } = this.props;
    const { show, hide } = this.state;

    const showClass = show ? 'show' : '';
    const hideClass = hide ? 'hide' : '';
    const classes = `alert-panel__alert alert-panel__alert--${type} ${showClass} ${hideClass}`;

    return <div className={classes}>{children}</div>;
  }
}

Alert.propTypes = {
  type: PropTypes.string,
  children: PropTypes.oneOfType([
    PropTypes.arrayOf(PropTypes.node),
    PropTypes.node,
  ]).isRequired,
};

export default Alert;
