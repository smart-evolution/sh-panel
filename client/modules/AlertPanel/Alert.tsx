import React from 'react';

type Props = {
  type: string;
  children: React.ReactNode;
};

type State = {
  show: boolean;
  hide: boolean;
};

class Alert extends React.Component<Props, State> {
  constructor(props: Props) {
    super(props);

    this.state = {
      show: false,
      hide: false,
    };
  }

  componentDidMount() {
    // @ts-ignore - to be fixed
    this.showTimeout = setTimeout(() => {
      this.setState({ show: true });
    }, 2000);
    // @ts-ignore - to be fixed
    this.hideTimeout = setTimeout(() => {
      this.setState({ hide: true });
    }, 4000);
  }

  componentWillUnmount() {
    // @ts-ignore - to be fixed
    if (this.showTimeout) {
      // @ts-ignore - to be fixed
      clearTimeout(this.showTimeout);
    }
    // @ts-ignore - to be fixed
    if (this.hideTimeout) {
      // @ts-ignore - to be fixed
      clearTimeout(this.hideTimeout);
    }
  }

  render() {
    const { type, children } = this.props;
    const { show, hide } = this.state;

    const showClass = show ? 'show' : '';
    const hideClass = hide ? 'hide' : '';
    const classes = `alert-panel__alert alert-panel__alert--${type} ${showClass} ${hideClass}`;

    return <div className={classes}>{children}</div>;
  }
}

export default Alert;
