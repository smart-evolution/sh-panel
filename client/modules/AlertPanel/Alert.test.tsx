import React from 'react';
import { mount } from 'enzyme';
import Alert from './Alert';

describe('AlertPanel/Alert', () => {
  it('should render correctly', () => {
    const component = mount(<Alert type="type-class">Message</Alert>);

    expect(component).toMatchSnapshot();
  });
});
