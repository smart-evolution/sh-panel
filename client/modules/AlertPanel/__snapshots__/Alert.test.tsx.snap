// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AlertPanel/Alert should render correctly 1`] = `
<Alert
  type="type-class"
>
  <div
    className="alert-panel__alert alert-panel__alert--type-class  "
  >
    Message
  </div>
</Alert>
`;
